internal/proxy/proxy.go

```diff
@@ -122,3 +128,29 @@ func (p *Proxy) Start(ctx context.Context, update Updater) {
 		}()
 	})
 }
+
+// newReverseProxy creates a configured httputil.ReverseProxy with common settings.
+func newReverseProxy(srv *Server, log *slog.Logger) *httputil.ReverseProxy {
+	return &httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL.Scheme = srv.Url.Scheme
+			request.URL.Host = srv.Url.Host
+			request.URL.Path = srv.Url.Path + request.URL.Path
+			request.Host = srv.Url.Host
+
+			log.Info("proxying request", "method", request.Method, "target", request.URL, "source", request.URL)
+		},
+		ModifyResponse: func(response *http.Response) error {
+			// Remove CORS headers from proxied response since we handle them in middleware
+			response.Header.Del("Access-Control-Allow-Origin")
```

Comment:
i think list of those headers should be aligned with same list in the middleware, getting them hardcoded makes things difficult to debug